src/main/kotlin/com/nikitanazarov/coroutinestacks/CoroutineStacksPanel.kt

```diff
@@ -1,64 +1,147 @@
-                    })
+                    session.process.addDebugProcessListener(panelBuilderListener())
                 }
 
                 override fun sessionRemoved(session: DebuggerSession) {
                     emptyText.text = CoroutineStacksBundle.message("no.java.debug.process.is.running")
                     emptyText.component.isVisible = true
-                    remove(coroutineGraph)
+                    panelContent.removeAll()
                 }
             })
     }
 
-    private fun buildCoroutineGraph(suspendContext: SuspendContext) {
-        // CoroutineDebugProbesProxy(suspendContext).dumpCoroutines()
-
-        val graph = mxGraph()
-        with(graph) {
-            val parent = defaultParent
-            model.beginUpdate()
-            try {
-                val v1 = insertVertex(parent, null, "Hello,", 20.0, 20.0, 80.0, 30.0)
-                val v2 = insertVertex(parent, null, "World!", 200.0, 150.0, 80.0, 30.0)
-                insertEdge(parent, null, "", v1, v2)
-            } finally {
-                model.endUpdate()
+    private fun panelBuilderListener() = object : DebugProcessListener {
+        override fun paused(suspendContext: SuspendContext) {
+            emptyText.component.isVisible = false
+            buildCoroutineGraph(suspendContext)
+        }
+
+        override fun resumed(suspendContext: SuspendContext?) {
+            panelContent.removeAll()
+        }
+    }
+
+    fun buildCoroutineGraph(suspendContext: SuspendContext) {
+        val suspendContextImpl = suspendContext as? SuspendContextImpl ?: run {
+            emptyText.text = CoroutineStacksBundle.message("coroutine.stacks.could.not.be.built")
+            return
+        }
+
+        val coroutineInfoCache = CoroutineDebugProbesProxy(suspendContextImpl).dumpCoroutines()
+        val coroutineInfoDataList = coroutineInfoCache.cache
+        val dispatcherToCoroutineDataList = mutableMapOf<String, MutableList<CoroutineInfoData>>()
+        for (info in coroutineInfoDataList) {
+            val dispatcher = info.descriptor.dispatcher ?: continue
+            dispatcherToCoroutineDataList.computeIfAbsent(dispatcher) { mutableListOf() }.add(info)
+        }
+
+        val dispatcherLabel = JLabel(CoroutineStacksBundle.message("select.dispatcher"))
+        val coroutineStacksPanelHeader = Box.createHorizontalBox()
+        val dispatchers = dispatcherToCoroutineDataList.keys.toTypedArray()
+        val dispatcherDropdownMenu = ComboBox(dispatchers)
+
+        dispatcherDropdownMenu.addActionListener {
+            val selectedDispatcher = dispatcherDropdownMenu.selectedItem as? String
+            val coroutineDataList = dispatcherToCoroutineDataList[selectedDispatcher]
+            if (!coroutineDataList.isNullOrEmpty()) {
+                updateCoroutineStackForest(coroutineDataList, suspendContextImpl)
+            }
+        }
+
+        val comboBoxSize = Dimension(Constants.comboBoxHeight, Constants.comboBoxWidth)
```

Comment:
Let's create a constant in a companion object of this class. 
```
val comboBoxSize = Dimension(150, 25)
```
And let's remove Constnants.comboBoxHeight and width